src/ParallelAlgorithms/Events/ObserveAtExtremum.hpp

```diff
@@ -0,0 +1,412 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <optional>
+#include <pup.h>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/ObservationBox.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "IO/Observer/GetSectionObservationKey.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ArrayIndex.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Local.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/Events/Tags.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/OptionalHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace detail {
+
+// Functional called by the reduction opearation: given 2 vectors,
+// return the vector with the largest first component
+struct MaxOfFirstElement {
```

Comment:
Move this into a more descriptive namespace, although it is very close to `funcl::Max` so actually you can probably just use that instead.

Reply:
Is this what funcl::Max would do when given a std::vector? I want the decision of which vector to keep to be solely based on the first element, carrying the other element along (I'm not sure what the behavior of funcl::Max is in that case)

Reply:
`funcl::Max` wraps `std::max` (the two-argument version), so it relies on the comparison operators for `std::vector`.  Those do a lexicographical comparison, which is the same as just comparing the first element unless there's a tie.

Reply:
Ok, great, should make this easier.